src/dict/dictionary.cc

```diff
@@ -116,25 +116,32 @@ void DataRequest::appendDataSlice( const void * buffer, size_t size )
   data.resize( data.size() + size );
 
   memcpy( &data.front() + offset, buffer, size );
+  cond.wakeAll();
 }
 
 void DataRequest::appendString( std::string_view str )
 {
   QMutexLocker _( &dataMutex );
   data.reserve( data.size() + str.size() );
   data.insert( data.end(), str.begin(), str.end() );
+  cond.wakeAll();
 }
 
 void DataRequest::getDataSlice( size_t offset, size_t size, void * buffer )
 {
-  if ( size == 0 )
-    return;
-
   QMutexLocker _( &dataMutex );
 
+  if ( size == 0 ) {
+    cond.wait( &dataMutex, 10 );
+    return;
```

Comment:
There are chances that the code will reach here.
wait on an unreached signal.

the timeout here is just a workaround.  need more investigation.